modules/ROOT/pages/understanding-orchestration-using-mule.adoc

```diff
@@ -0,0 +1,62 @@
+= Understanding Orchestration Using Mule
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: anypoint studio, studio, mule esb, orchestration
+
+Orchestration is how Mule facilitates the process of moving data through a Mule implementation based on flows. Orchestration can be seen as combining service calls to create higher-level, more useful composite services, and implies implementing business-level processes combining business-specific services across applications and information systems.
+
+== Orchestration and Flows
+
+Orchestration involves decoupling point-to-point integration in favor of chains of messages sent to sources that can be reused or changed as systems or business needs change. It involves moving beyond patterns into a service-based framework, where you wrap multiple components with sources to decouple them from protocols and routing requirements, enabling multiple services to work as one when implementation requirements demand it. Some common use cases that flows easily orchestrates include simple integration, scheduled data processing, connecting cloud and on-premise applications, and event processing where multiple services must be composed and operate simultaneously.
+
+== How Flows Enables Implementation of Orchestration
+
+Flows enables new ways to orchestrate service mediation—for example, you can combine an arbitrary number of sources, routers, and components that work in concert to achieve implementation requirements. Flows enable you to orchestrate new forms of data movement based on flow-specific routers. Orchestration enables you to create and reuse subflows, chain together multiple components in one flow, and integrate sources or connectors anywhere in a flow.
+
+== When to Use Orchestration
+
+You use orchestration when you need to combine existing services to meet business requirements. For example, use orchestration when you're adding a new employee who needs to be entered in both the payroll system and the HR system. In another example of when to use orchestration, you might use it when you accept an order, make sure the order is in the inventory system, use the credit check system to ensure you trust the customer's credit, then put the order in the billing system, and finally, put the order in the shipping system.
+
+== An Example of Orchestration
+
+The following example of orchestration uses a flow to accept a book order in an XML file and transform the file’s contents into a standard format that contains one or more specific orders. Next, the orders are split out to be processed separately, and any non-book orders are filtered out. Now two services are called: one to check whether the book is in our inventory, and another to create a customer order. This is both e-mailed to the customer and stored in a database. If anything in this process fails, the message is placed in a JMS queue of failed orders, where it can be checked, any error fixed, and finally resubmitted.
```

Comment:
I suggest using numbered lists for the process steps:

The following example of orchestration does the following:

Uses a flow to accept a book order in an XML file and transforms the file’s contents into a standard format that contains one or more specific orders. 

Splits the orders out to be processed separately.

Filters out non-book orders. (_Does this step happen before the last one?_)

Calls two services: one to check whether the book is in the inventory, and another to create a customer order. Theo order is both e-mailed to the customer and stored in a database. If anything in this process fails (_which process - the this step or the whole process?_), the message is placed in a JMS queue of failed orders, where you can determine and fix the issue, and resubmit.

Reply:
Just checking - Does the orchestration encompass all of these steps or just the last one? My text assumes it encompasses the whole process.

Reply:
This is fine to add.